dtfs-central-api/src/repositories/tfm-deals-repo/tfm-deal-cancellation.repo.ts

```diff
@@ -109,4 +110,43 @@ export class TfmDealCancellationRepo {
 
     return updateResult;
   }
+
+  /**
+   * submits the deal cancellation and updates the respective deal stage
+   * @param dealId - The deal id
+   * @param cancellation - The deal cancellation details to submit
+   * @param auditDetails - The users audit details
+   */
+  public static async submitDealCancellation(
+    dealId: string | ObjectId,
+    cancellation: TfmDealCancellation,
+    auditDetails: AuditDetails,
+  ): Promise<TfmDealCancellationResponse> {
+    if (!ObjectId.isValid(dealId)) {
+      throw new InvalidDealIdError(dealId.toString());
+    }
+
+    const dealCollection = await this.getCollection();
+
+    const updateDeal = await dealCollection.updateOne(
+      {
+        _id: { $eq: new ObjectId(dealId) },
+        'tfm.stage': { $ne: TFM_DEAL_STAGE.CANCELLED },
+        'dealSnapshot.submissionType': { $in: [DEAL_SUBMISSION_TYPE.AIN, DEAL_SUBMISSION_TYPE.MIN] },
+        'tfm.cancellation.reason': { $eq: cancellation.reason },
+        'tfm.cancellation.bankRequestDate': { $eq: cancellation.bankRequestDate },
+        'tfm.cancellation.effectiveFrom': { $eq: cancellation.effectiveFrom },
+      },
```

Comment:
It's been a while since i've worked with MongoDB, I think this could just be this so we're purely matching the ID. ?

```ts
{
  _id: { $eq: new ObjectId(dealId) },
  'tfm.stage': { $ne: TFM_DEAL_STAGE.CANCELLED },
},
```

Reply:
Ah yes, so the reason for _also_ matching on the reason/bankRequestDate/effectiveFrom is just an extra bonus check that we are cancelling the deal with the data the user expects. 

It stops a very unlikely edge case:
- Say two people are editing the deal cancellation at the same time
- Say between Person 1 viewing the deal summary and pressing submit, Person 2 changes some of the data. 
- These 3 lines of code would stop the submission going through as the viewed data Person 1 sees no longer matches what's in the database.

Reply:
Thanks @BethThomas141 seems slightly overkill IMO, but good to prevent the edge case.